apps/web/src/actions/datasets/create.ts

```diff
@@ -1,41 +1,88 @@
 'use server'
 
+import { DatasetsRepository } from '@latitude-data/core/repositories'
 import { createDataset } from '@latitude-data/core/services/datasets/create'
 import disk from '$/lib/disk'
 import { z } from 'zod'
 
 import { authProcedure } from '../procedures'
 
+const DELIMITERS_KEYS = [
+  'comma',
+  'semicolon',
+  'tab',
+  'space',
+  'custom',
+] as const
+const DELIMITER_VALUES = {
+  comma: ',',
+  semicolon: ';',
+  tab: '\t',
+  space: ' ',
+}
+
 const MAX_SIZE = 3
 const MAX_UPLOAD_SIZE_IN_MB = 3 * 1024 * 1024
-
 export const createDatasetAction = authProcedure
   .createServerAction()
   .input(
-    z.object({
-      name: z.string().min(1, { message: 'Name is required' }),
-      dataset_file: z
-        .instanceof(File)
-        .refine((file) => {
-          return !file || file.size <= MAX_UPLOAD_SIZE_IN_MB
-        }, `Your dataset must be less than ${MAX_SIZE}MB in size`)
+    async ({ ctx }) => {
+      return z
+        .object({
+          name: z
+            .string()
+            .min(1, { message: 'Name is required' })
+            .refine(
+              async (name) => {
+                const scope = new DatasetsRepository(ctx.workspace.id)
+                const existing = await scope.findByName(name)
+                return !existing
+              },
+              {
+                message:
+                  'This name was already used, plese use something different',
+              },
+            ),
+          csvDelimiter: z.enum(DELIMITERS_KEYS, {
+            message: 'Choose a valid delimiter option',
+          }),
+          csvCustomDelimiter: z.string(),
```

Comment:
Shouldn't `csvCustomDelimiter` be `optional`?

Reply:
It's ok, we always send an empty string because the field is always sent. Is in the validation that we check user picked custom and this field has content